src/mpid/ch4/shm/posix/posix_init.c

```diff
@@ -83,23 +83,24 @@ int MPIDI_POSIX_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag
     MPIDI_POSIX_global.am_buf_pool =
         MPIDIU_create_buf_pool(MPIDI_POSIX_BUF_POOL_NUM, MPIDI_POSIX_BUF_POOL_SIZE);
 
-    MPIR_CHKPMEM_DECL(3);
+    MPIR_CHKPMEM_DECL(1);
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_INIT_HOOK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_INIT_HOOK);
 
     /* Populate these values with transformation information about each rank and its original
      * information in MPI_COMM_WORLD. */
 
-    mpi_errno = MPIR_Find_local(MPIR_Process.comm_world, &num_local, &my_local_rank,
-                                /* comm_world rank of each local process */
-                                &MPIDI_POSIX_global.local_procs,
-                                /* local rank of each process in comm_world if it is on the same node */
-                                &MPIDI_POSIX_global.local_ranks);
-
-    local_rank_0 = MPIDI_POSIX_global.local_procs[0];
-    MPIDI_POSIX_global.num_local = num_local;
-    MPIDI_POSIX_global.my_local_rank = my_local_rank;
+    MPIDI_POSIX_global.local_procs = MPIR_Process.node_local_map;
+    MPIDI_POSIX_global.local_ranks = (int *) MPL_malloc(MPIR_Process.size * sizeof(int),
+                                                        MPL_MEM_SHM);
+    memset(MPIDI_POSIX_global.local_ranks, -1, MPIR_Process.size);
```

Comment:
`memset` sets "bytes". Need explicit "for-loop" here. You may consider combine with the following loop.